File: domain/src/main/java/com/insurtech/kanguro/domain/model/Pet.kt
package com.insurtech.kanguro.domain.model

import android.os.Parcelable
import com.insurtech.kanguro.common.enums.Gender
import com.insurtech.kanguro.common.enums.PetSize
import com.insurtech.kanguro.common.enums.PetType
import kotlinx.parcelize.Parcelize
import java.util.*

@Parcelize
data class Pet(
    val id: Long? = null,
    val name: String? = null,
    val birthDate: Date? = null,
    val isBirthDateApproximated: Boolean? = null,
    val gender: Gender? = null,
    val type: PetType? = null,
    val size: PetSize? = null,
    val isNeutered: Boolean? = null,
    val preExistingConditions: String? = null,
    val petBreedId: Long? = null,
    val breed: String? = null,
    val additionalInfoSessionId: String? = null,
    val hasAdditionalInfo: Boolean? = null,
    val petPictureUrl: String? = null,
    val pictureId: Int? = null
) : Parcelable

File: iosApp/Modules/KanguroPet/KanguroPetDomain/Sources/KanguroPetDomain/Models/Pet.swift
import Foundation
import KanguroSharedDomain

public struct Pet: Equatable {
    public var id: Int
    public var name: String?
    public var birthDate: Date?
    public var isBirthDateApproximated: Bool?
    public var gender: PetGender?
    public var type: PetType?
    public var size: String?
    public var isNeutered: Bool?
    public var preExistingConditions: String?
    public var petBreedId: Int?
    public var breed: String?
    public var zipCode: String?
    public var territoryId: Int?
    public var additionalInfoSessionId: String?
    public var hasAdditionalInfo: Bool?
    public var petPictureUrl: String?
    public var cloudDocumentPolicies: [CloudDocumentPolicy]?

    public init(
        id: Int,
        name: String? = nil,
        birthDate: Date? = nil,
        isBirthDateApproximated: Bool? = nil,
        gender: PetGender? = nil,
        type: PetType? = nil,
        size: String? = nil,
        isNeutered: Bool? = nil,
        preExistingConditions: String? = nil,
        petBreedId: Int? = nil,
        breed: String? = nil,
        zipCode: String? = nil,
        territoryId: Int? = nil,
        additionalInfoSessionId: String? = nil,
        hasAdditionalInfo: Bool? = nil,
        petPictureUrl: String? = nil,
        cloudDocumentPolicies: [CloudDocumentPolicy]? = nil
    ) {
        self.id = id
        self.name = name
        self.birthDate = birthDate
        self.isBirthDateApproximated = isBirthDateApproximated
        self.gender = gender
        self.type = type
        self.size = size
        self.isNeutered = isNeutered
        self.preExistingConditions = preExistingConditions
        self.petBreedId = petBreedId
        self.breed = breed
        self.zipCode = zipCode
        self.territoryId = territoryId
        self.additionalInfoSessionId = additionalInfoSessionId
        self.hasAdditionalInfo = hasAdditionalInfo
        self.petPictureUrl = petPictureUrl
        self.cloudDocumentPolicies = cloudDocumentPolicies
    }
}
